### PR TITLE
Homework 5

### PR DESCRIPTION
**Pull request by Zhandoszzz**
Server with FastAPI with following methods:
1. Add molecule (smiles) and its identifier
2. Get molecule by identifier
3. Updating a molecule by identifier
4. Delete a molecule by identifier
5. List all molecules
6. Substructure search for all added molecules
7. Upload file with molecules (txt file).

### CODE REVIEW

File: src/main.py [hunk 1]
```diff
@@ -1,7 +1,80 @@
+from pydantic import BaseModel
 from rdkit import Chem
+from fastapi import FastAPI, HTTPException, status, File, UploadFile
+import json
 
 
-def substructure_search(mols, mol):
+app = FastAPI()
+molecules_db = []
```
Comment (JDima) [inline src/main.py]: The choice of data structure list is not good idea, since most queries will run in linear time, better to choose dictionary

**Comment by JDima** [comment]
Looks good! In the future, when implementing API, try to write comments and handle errors just as well. Pay attention to the choice of structure for storing data, since now queries will work quite a significant time with large data. Try to fix and merge

File: src/main.py [hunk 2]
```diff
@@ -1,7 +1,80 @@
...
+            return updated_mol
+    raise HTTPException(status_code=status.HTTP_404_NOT_FOUND,
+                        detail='Molecule Not Found')
+
+
+@app.delete("/molecules/{mol_id}", status_code=status.HTTP_204_NO_CONTENT)
+def delete_molecule_by_id(mol_id: int):
+    for index, mol in enumerate(molecules_db):
+        if mol['id'] == mol_id:
+            molecules_db.pop(index)
```
Comment (JDima) [inline src/main.py]: Here `pop` will work linear time, as result `O(N^2)` for function.

Comment (Zhandoszzz) [reply src/main.py]: Thank you for your feedback! I Changed storage to dictionary for faster access by id and will merge

**Pull request merged**